src/edge_kernels/MomentumEdgeSolverAlg.C

```diff
@@ -146,12 +160,36 @@ MomentumEdgeSolverAlg::execute()
         }
       }
 
+      // Upwinding switch for multiphase cases.
+      // Factors determined by ensuring full upwinding
+      // at interfaces with interface
+      // widths that are ~2 cells thick
+      DblType density_upwinding_factor = 1.0;
+      DblType alphaUpw_w_vof = alphaUpw;
+      DblType om_alphaUpw_w_vof = 1.0 - alphaUpw_w_vof;
```

Comment:
We get so many questions about upwinding in nalu-wind and all the settings. It feels like we should document these extra modifications?

Reply:
Along those lines. I would be in favor of adding a warning or throwing an error if the upwinding is put in a state that you have discovered is unstable. I'm assuming pure central will not work with VOF? Or is you modification below sufficient to ensure that the local upwinding is sufficient?  Not knowing is also an acceptable answer, but I would like a clear guide line for the best practice, preferably  in the code that is checked at runtime.

Reply:
The documentation should definitely reflect the changes here. Our general finding is that pure central at the multiphase interface does not work well for VOF when you have high density ratios and strong flow dynamics. 

The modifications here respect the users settings except explicitly at the multiphase interface when VOF is turned on. I think, at best, an appropriate warning would be to inform the user that by enabling VOF you have implicitly turned on upwinding at the interface.

Reply:
Warning has been added.